tests/grids/test_grid.py

```diff
@@ -132,6 +132,218 @@ def test_cell_faces_as_dense(g: pp.Grid):
+
+        Returns:
+            A list of grids.
+
+        """
+        g1 = pp.CartGrid([2, 2], [1, 1])
+        g2 = pp.CartGrid([2, 2, 2], [1, 1, 1])
+        g3 = pp.StructuredTriangleGrid([2, 2], [1, 1])
+        g4 = pp.StructuredTetrahedralGrid([2, 2, 2], [1, 1, 1])
+
+        x = np.array([[0, 0.5, 0.5, 1], [0, 0, 0, 0], [0, 0, 0, 0]])
+        fn = sps.identity(4, format="csr")
+        cf = sps.csc_matrix(
+            np.array([[-1, 1, 0, 0], [0, 0, -1, 1]]).T
+        )
+        g5 = pp.Grid(dim=1, nodes=x, face_nodes=fn, cell_faces=cf, name="")
+
+        grid_list = [g1, g2, g3, g4, g5]
+
+        for g in grid_list:
+            g.compute_geometry()
+        return grid_list
+
+    def _boundary_faces(self, g: pp.Grid) -> np.ndarray:
+        """Find the boundary faces of the grid. 
+        
+        The boundary faces are identified by geometrically checking if the face center
+        is at the boundary of the domain, which is assumed to be of unit size. For 1d
+        grids, we also know that faces with center at x=0.5 are boundary faces. This
+        construction is not general, but it allows us to find the boundary faces without
+        using boundary tags and similar, and thus provide a real check of the
+        implementation of the divergence and trace operators.
+
+        Parameters:
+            g: The grid for which the boundary faces should be found.
+            
+        Returns:
+            np.ndarray: The indices of the boundary faces.
+
+        """
+
+        bf = np.where(np.logical_or(np.any(g.face_centers[:g.dim] == 0, axis=0),
+             np.any(g.face_centers[:g.dim] == 1, axis=0)))[0]
+
+        if g.dim == 1:
+            bf = np.hstack((bf, np.where(g.face_centers[0] == 0.5)[0]))
+        
+        bf.sort()
+        return bf
+    
+
+    def _bound_int_ind(self, g: pp.Grid, dim: int) -> tuple[np.ndarray, np.ndarray]:
+        """Find the indices correspodning to  boundary and internal faces of a grid.
+
+        The indices are expanded to account for the dimension 'dim' of an operator (e.g.
+        the divergence operator).
+
+        Parameters:
+            g: The grid for which the indices should be found.
+            dim: The dimension of the operator.
+
+        Returns:
+            tuple with the indices corresponding to the boundary and internal faces.
+
+        """
+
+        # Faces on the boundary, and the indices of these faces in the divergence
+        # matrix. These will be identical if dim=1, but for dim > 1, there will be dim
+        # boundary indices for each boundary face.
+        bound_faces = self._boundary_faces(g)
+        bound_ind = pp.fvutils.expand_indices_nd(bound_faces, dim)
+        # The indices corresponding to internal faces are the complement of the boundary
+        # indices.
+        int_face_ind = np.setdiff1d(np.arange(g.num_faces * dim), bound_ind)
+
+        return bound_ind, int_face_ind
+
+
+    @pytest.mark.parametrize("dim", [1, 2])
```

Comment:
I think we should do dim=3 as well.

Reply:
I don't see how that will increase the quality of the test; the only fork in the code to be tested is 1 vs many and if the ordering of indices somehow gets wrong for dim=3, it will almost surely also do so for dim=2. By all means, we can add it, but I don't think the extra time for testing is warranted.